lib/src/components/dropdown/dropdown.dart

```diff
@@ -0,0 +1,440 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// Class for [ZetaDropDown]
+class ZetaDropDown extends StatefulWidget {
+  ///Constructor of [ZetaDropDown]
+  const ZetaDropDown({
+    super.key,
+    required this.items,
+    required this.onChange,
+    required this.selectedItem,
+    this.rounded = true,
+    this.checkBoxType = CheckBoxType.none,
+    this.isLarge = false,
+  });
+
+  /// Input items as list of [ZetaDropdownItem]
+  final List<ZetaDropdownItem> items;
+
+  /// Currently selected item
+  final ZetaDropdownItem selectedItem;
+
+  /// Handles changes of dropdown menu
+  final ValueSetter<ZetaDropdownItem> onChange;
+
+  /// {@macro zeta-component-rounded}
+  final bool rounded;
+
+  /// If checkbox is to be shown and if its circular or square
+  final CheckBoxType checkBoxType;
+
+  /// If menu is large or minimised.
+  final bool isLarge;
```

Comment:
Can we change this to instead be called `size` and either make a ZetaDropdownSize enum or use the existing ZetaWidgetSize type? @thelukewalton  what do you think?

Reply:
It seems the design only accounts two sizes being large and minimized, so i thought having a boolean would be a better approach

Reply:
I think this is fine considering there are only 2 options - this is what we have done elsewhere. Perhaps rather than large / minimised it should be open / closed?

Reply:
Just checked Figma uses the word mini, so maybe it should be `isMini`?

Reply:
I've altered to match the design so now it can either be default or minimized